### PR TITLE
adds another component for heatmap

### PR DESCRIPTION
**Pull request by rishabgupta94**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -19,15 +19,21 @@ jobs:
       - name: Set up Node.js
         uses: actions/setup-node@v3
         with:
-          node-version: "18.20.2"
+          node-version: "22.8.0"
```
Comment (rishabgupta94) [inline .github/workflows/deploy.yml:22]: is this the node version that you are using? Make sure they are the same to avoid any surprise issues.

File: package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
     "preview": "vite preview",
     "predeploy": "npm run build",
-    "deploy": "gh-pages -d dist"
+    "deploy": "gh-pages -d dist -r https://x-access-token:${GITHUB_TOKEN}@github.com/jvpatey/OralCheckr.git"
```
Comment (rishabgupta94) [inline package.json:13]: I am not sure about this, but if this works, then it is fine.

File: src/containers/habit-tracker/analytics/year-view/Heatmap.tsx [hunk 3]
```diff
@@ -101,11 +80,6 @@ const useHeatmapOptions = (
       events: {
         mounted: () => {
```
Comment (rishabgupta94) [inline src/containers/habit-tracker/analytics/year-view/Heatmap.tsx]: we don't need this event anymore

File: src/containers/habit-tracker/analytics/year-view/Heatmap.tsx [hunk 4]
```diff
@@ -162,7 +136,7 @@ const useHeatmapOptions = (
     },
     yaxis: {
       title: {
-        text: windowWidth > 768 ? "Week days" : "", // Show 'Week days' on larger screens, hide on smaller screens
+        // text: windowWidth > 768 ? "Week days" : "", // Show 'Week days' on larger screens, hide on smaller screens
```
Comment (rishabgupta94) [inline src/containers/habit-tracker/analytics/year-view/Heatmap.tsx]: remove commented code

File: src/containers/habit-tracker/analytics/year-view/Heatmap.tsx [hunk 5]
```diff
@@ -205,66 +174,40 @@ interface HeatmapProps {
...
-    return () => {
-      window.removeEventListener("resize", handleResize);
-    };
-  }, []);
-
-  const handleRendered = () => {
-    console.log("Chart rendered, setting 'rendered' to true");
-    setRendered(true);
+  const isCompMounted = (isMounted: boolean) => {
+    console.log(isMounted);
```
Comment (rishabgupta94) [inline src/containers/habit-tracker/analytics/year-view/Heatmap.tsx]: remove log

File: src/containers/habit-tracker/analytics/year-view/Heatmap.tsx [hunk 6]
```diff
@@ -205,66 +174,40 @@ interface HeatmapProps {
...
 
-    return () => {
-      window.removeEventListener("resize", handleResize);
-    };
-  }, []);
-
-  const handleRendered = () => {
-    console.log("Chart rendered, setting 'rendered' to true");
-    setRendered(true);
+  const isCompMounted = (isMounted: boolean) => {
```
Comment (rishabgupta94) [inline src/containers/habit-tracker/analytics/year-view/Heatmap.tsx]: rename function to `isHeatmapChartMounted`.

Also, add a comment about why we are doing this extra step of checking the mounting. Explain the problem in brief in a comment.

File: src/containers/habit-tracker/analytics/month-view/MonthView.tsx [hunk 7]
```diff
@@ -157,47 +158,43 @@ export function MonthView({
   hideAnalytics,
 }: ViewProps) {
   const [selectedMonth, setSelectedMonth] = useState<Date>(new Date());
+  const [loading, setLoading] = useState(true);
+
+  // Function to set loading to false after the component is mounted
+  const isCompMounted = (isMounted: boolean) => {
```
Comment (rishabgupta94) [inline src/containers/habit-tracker/analytics/month-view/MonthView.tsx]: rename function. Same suggestion as in the HeatMap.
Also add comment on why is this needed.

File: src/containers/habit-tracker/analytics/month-view/MonthView.tsx [hunk 8]
```diff
@@ -157,47 +158,43 @@ export function MonthView({
...
+    console.log(isMounted);
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    console.log("Mounted");
```
Comment (rishabgupta94) [inline src/containers/habit-tracker/analytics/month-view/MonthView.tsx]: remove all console logs

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (jvpatey) [reply .github/workflows/deploy.yml:22]: Yes it is. I ended updated node between when I first made the deploy file and now.
